identity/aziot-identityd/src/lib.rs

```diff
@@ -386,7 +386,7 @@ impl Api {
                 // Clear the backed up device state before reprovisioning.
                 // If this fails, log a warning but continue with reprovisioning.
                 let mut backup_file = self.settings.homedir.clone();
-                backup_file.push("device_info");
+                backup_file.push(identity::DEVICE_BACKUP_LOCATION);
 
                 if let Err(err) = std::fs::remove_file(backup_file) {
```

Comment:
Need to delete the new files too?

Reply:
If /reprovision API is called, it means the user wants to wipe the device backup completely. The module backup for that device is purged only on setting the device with new provisioning information in IdentityManager. Even that is purged only when the device has changed.

Reply:
I'm asking why it makes sense to leave the module backup around when the device backup has been deleted. If the device hasn't been reprovisioned in the cloud the same backup will be written back at the end, so in the same way the same backup for the modules can also be written back.

Reply:
That is intentional to leave a backup if any provisioning / reconciliation fails. Two reasons- 1) IdentityManager always prioritizes and overwrites fresh identity data from the cloud for all operations and b) IdentityManager will cleanup the module identity backups on reconciliation. Leaving the module backup alone doesn't cause any issues (even though in some very corner cases, modules maybe left behind) because it is only accessible after a) the device is set in Identity Manager (after reprovisioning) and b) the "iothub + device" match. The purging should only happen when step (b) has a new device. Setting it prior to that isn't necessary. If we always deleted module backup before both (a) and (b) happen AND reconcile ever fails, the backup won't be available for /sign operations in some valid scenarios.